src/charm.py

```diff
@@ -45,23 +49,32 @@ class LokiOperatorCharm(CharmBase):
 
     _stored = StoredState()
     _port = 3100
-    _name = "loki"
+    _name = PEER
 
     def __init__(self, *args):
         super().__init__(*args)
         self._container = self.unit.get_container(self._name)
         self._stored.set_default(k8s_service_patched=False, config="")
         self.service_patch = KubernetesServicePatch(self, [(self.app.name, self._port)])
         self.alertmanager_consumer = AlertmanagerConsumer(self, relation_name="alertmanager")
+        self.ingress_per_unit = IngressPerUnitRequirer(
+            self, endpoint="ingress-per-unit", port=self._port
+        )
+        external_url = urlparse(self._external_url)
         self.grafana_source_provider = GrafanaSourceProvider(
             charm=self,
             refresh_event=self.on.loki_pebble_ready,
-            source_type="loki",
-            source_port=str(self._port),
+            source_type=PEER,
+            source_url=self._external_url,
+        )
+
+        self.loki_provider = LokiPushApiProvider(
+            self,
```

Comment:
I don’t understand why we don’t do self.hostname in the else branch

Reply:
"else branch"? What do you mean?

Reply:
See next comment